### ISSUE TITLE
[4.0] Backend: Plugin-update changelog fails

### ISSUE THREAD

**Issue by hgh-esn** [joomla/joomla-cms#25017]
**### Steps to reproduce the issue**
There must be a plugin installed,  witch has the necessary changelog-/server and -data activ. [See](https://github.com/roland-d/component_joomla/releases/tag/v0.0.2) 
Go into System>Update>Extensions. In the upcoming extension-update-listview the changelog-item fails 

**### Expected result**

There should be a link to a modal-windows at the changelog-item

**### Actual result**

 I get no "changelog-version-item-link". Instead of this, I see **N/A** or in German: **Unbekannt**
<img width="650" alt="2019-05-27_13h49_31" src="https://user-images.githubusercontent.com/5299726/58417878-7b0e0480-8086-11e9-9c06-d867ef650f5a.png">

**### System information (as much as possible)**

J4-alpha-dev-10 (NightlyBuild: 190525)

**### Additional comments**

**Comment by ReLater** [comment]
See Issue https://github.com/joomla/joomla-cms/issues/23063

**Comment by ghost** [comment]
@ReLater so its a known issue?

**Comment by ReLater** [comment]
> so its a known issue?

No. I was wrong. 2 different issues.

**Comment by hgh-esn** [comment]
@ReLater 
And when you or someone else will make a change to bring back the origin functionality?

**Comment by ReLater** [comment]
I can confirm this issue. [I've build a custom plugin](https://github.com/ReLater/plg_system_changelogtest/releases) and followed the documentation https://docs.joomla.org/Adding_changelog_to_your_manifest_file

The updateserver infos are: https://raw.githubusercontent.com/ReLater/plg_system_changelogtest/master/updateserver-infos.xml

The cahngelog infos are https://raw.githubusercontent.com/ReLater/plg_system_changelogtest/master/changelog-infos.xml

After installing of version 0.0.1 of my test plugin.

Everything's OK in view "Extensions Manage"

![30-05-_2019_12-18-53](https://user-images.githubusercontent.com/20780646/58626630-899e2b00-82d5-11e9-97e3-cb6e8bd84d8c.jpg)

but not in view "Extensions: Update":

![30-05-_2019_12-18-03](https://user-images.githubusercontent.com/20780646/58626765-d7b32e80-82d5-11e9-94d8-093c72b14b50.jpg)

After tests with my own plugin I tested with:
https://github.com/roland-d/component_joomla/releases/tag/v0.0.2

There the changelog in view "Extensions: Update" is empty:
![30-05-_2019_12-44-02](https://user-images.githubusercontent.com/20780646/58627934-b30c8600-82d8-11e9-854f-69a6da2cf40e.jpg)

@roland-d (creator of pr https://github.com/joomla/joomla-cms/pull/24026

> And when you or someone else will make a change to bring back the origin functionality?

Sorry, I have no idea how.

**Comment by ReLater** [comment]
Addition:
After installation of my plugin field `changelogurl` in database table `#__updates` is empty (NULL). 

![30-05-_2019_12-57-03](https://user-images.githubusercontent.com/20780646/58628826-08499700-82db-11e9-9668-7bbf09afb038.jpg)

When I add it manually I see the Changelog button but the update changelog for version 0.0.2 is empty, too.

![30-05-_2019_13-02-13](https://user-images.githubusercontent.com/20780646/58628939-56f73100-82db-11e9-89a7-bf228e8d3092.jpg)

**Comment by ReLater** [comment]
Addition 2:

I sse now 2 Changelog buttons. But both open the same empty changelog of my plugin version 0.0.2

![30-05-_2019_13-06-41](https://user-images.githubusercontent.com/20780646/58629184-0cc27f80-82dc-11e9-8a4e-e90d57cac404.jpg)

And when I make a database fix for `Joomla CMS` my manually added `changelogurl` is removed from database table `#__updates.` Somehow weird. At least too weird for me ;-)

**Comment by roland-d** [comment]
I will take a look

**Comment by roland-d** [comment]
This is also related to PR #24890 which I need to fix again due to the conflict.

@ReLater There is an issue with your manifest file. This manifest file: https://raw.githubusercontent.com/ReLater/plg_system_changelogtest/master/updateserver-infos.xml has no `changelogurl` tag in it. That is why it is missing from the updates table.

See my test manifest file here: https://rolandd.com/24026/lists.xml

@hgh-esn I think you have the same issue.

As far as I can tell, nothing is broken, just the manifest file is missing the changelogurl node.

**Comment by roland-d** [comment]
Now that #24890 is merged, you may want to update your test installation as well. This contains some important fixes.

**Comment by ReLater** [comment]
> There is an issue with your manifest file. This manifest file: updateserver-infos.xml has no changelogurl tag in it. That is why it is missing from the updates table.

I tried it without and with tag but then had the empty popup and was unsure... I've added an example code to the documentation to clarify which manifest file we talk about: 
https://docs.joomla.org/Adding_changelog_to_your_manifest_file

**Comment by roland-d** [comment]
The empty popup is because of issue #24890 It should work after you have those changes. In case you use the nightly build, you will have to wait till tomorrow, otherwise you need to run `npm run build:js` from the root folder.

**Comment by ReLater** [comment]
Thanks! Will wait for the next nightly.

**Comment by hgh-esn** [comment]
@roland-d

Thanks for your quick work - I'll wait also until the next nightly tomorrow.

> ..... I think you have the same issue.
I have this issue , but I have the `changelogurl tag` in my manifest-file.

**Comment by ReLater** [comment]
Tested successfully. All issues are solved.
Thank you!

**Comment by hgh-esn** [comment]
@roland-d
sorry, I could get it to run, but I haven't any content in the popping up modal.
I miss the `oncklick="(...)"` in the` /administrator/components/com_installer/tmpl/update/default.php`
Is that right? 

@ReLater 
Do you have content in the modal?

**Comment by roland-d** [comment]
@hgh-esn Make sure you have the last nightly build. This contains all the needed changes to make it work.

**Comment by ReLater** [comment]
> Do you have content in the modal?

Yes. All worked as expected after update to before-last nightly.
Maybe you want to test and check my plugin https://github.com/ReLater/plg_system_changelogtest . All relevant files can be seen in the master branch.
Install release 0.0.1!

Only the close buttons of the popups don't work corecctly but as far as I see that's a CSS issue.
![01-06-_2019_13-30-40](https://user-images.githubusercontent.com/20780646/58747889-8c358780-8471-11e9-85ff-7f9e7c7da33e.jpg)

**Comment by hgh-esn** [comment]
@ReLater @roland-d 

I got it to run now. But it was something strange.
For my tests of the earlier version, I renamed the file: .../changelog.xml    ->     .../changelog**s**.xml
 - I found this name better, because of having more than one `<update>-tags` in there. So (with '**s**') it was in the database after the plugin-installation..

To test the latest version I haven't deinstalled my example plugin. But the xml-filename now was changed back - by default - to` .../changelog.xml` without '**s**'. So it doesn't match to the db-entry and I only saw the modal with it's header line and a "blank" content area..

There was no message for the mismatching filenames.

@roland-d 
It will be great, if you will give out an alert in the modal or as joomla-message, if there is a conflict between the filenames in the `<updateurl-tag>` of the xml-file and that, what is stored in the database.

@ReLater
The close button of the modal is ok in my tests as you can see here..
<img width="471" alt="2019-06-04_03h03_07" src="https://user-images.githubusercontent.com/5299726/58844011-85776200-8675-11e9-97af-0a6924a40896.png">

@roland-d 
Just another effect.
When you stay at an update-screen and the session becomes a timeout, you will see this message in the modal, when clicking to the version-link  or update-button on the stayed screen  - without reacting Joomla by a new login.:
<img width="467" alt="2019-06-04_03h20_04" src="https://user-images.githubusercontent.com/5299726/58844597-40086400-8678-11e9-91cc-4ca89b4f5193.png">

That is a bit cryptical to users. Is there a way to bring out a more user readable message? 

And last:
Where is that js-stored in the joomla file-structure?

 
 

 .

**Comment by roland-d** [comment]
@hgh-esn I find it hard to follow what you are doing.

> It will be great, if you will give out an alert in the modal or as joomla-message, if there is a conflict between the filenames in the <updateurl-tag> of the xml-file and that, what is stored in the database.

If you manually change the file on your server, you cannot expect Joomla to know what file you have there. If you want an updated changelog file to be read, you will need to re-install the extension.

> That is a bit cryptical to users. Is there a way to bring out a more user readable message?

Please create a new issue for it so we can look into it. This issue is closed.

> Where is that js-stored in the joomla file-structure?

build/media_source/com_installer/js/changelog.es6.js

**Comment by 810** [comment]
Hi, can this issue open again, because on install the changelogurl is again empty with `<changelogurl>` on xml.

**Comment by Quy** [comment]
@810 Is this the same issue as #29037?

**Comment by ReLater** [comment]
> Hi, can this issue open again, because on install the changelogurl is again empty with <changelogurl> on xml.

Is this related anyhow? https://github.com/joomla/joomla-cms/issues/29037

**Comment by 810** [comment]
No, on clean new install the plugin changelogurl isn't saved on the database on _extensions. other types (package/component/media/library) are working.

**Comment by zero-24** [comment]
> Hi, can this issue open again, because on install the changelogurl is again empty with <changelogurl> on xml.

Well what about opening a new issue and add a link there? As this issue was closed 31 May 2019 nearly one year ago?
